hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/Mutiny.java

```diff
@@ -756,9 +756,25 @@
 	interface SessionFactory extends AutoCloseable {
 
 		/**
-		 * Obtain a new {@link Session reactive session}, the
-		 * main interaction point between the user's program and
-		 * Hibernate Reactive.
+		 * Obtain a new {@link Session reactive session}, the main
+		 * interaction point between the user's program and Hibernate
+		 * Reactive.
+		 *
+		 * The underlying database connection is obtained lazily
+		 * when the returned {@link Session} needs to access the
+		 * database.
+		 *
+		 * The client must close the session using {@link Session#close()}.
+		 */
+		Session createSession();
+
+		/**
+		 * Obtain a new {@link Session reactive session}, the main
+		 * interaction point between the user's program and Hibernate
+		 * Reactive.
+		 *
+		 * The underlying database connection is obtained before the
+		 * {@link Session} is returned via a {@link Uni}.
```

Comment:
Since `openSession()` also retrieves a connection lazily, I wonder if we still need this method at all? We can keep them both around for now, but by first release we may want to pick one or the other.  Obviously I prefer `createSession()` (because I requested it) and users can pretty easily wrap the `Session` in a `Uni` on their own if they want

Reply:
Well `openSession()` is good for people who might want to report a connection failure in a specific way.